### PR TITLE
Added info log for datafile revision update.

### PR DESCRIPTION
**Pull request by yasirfolio3**
1. Added datafile revision updated log in datafile polling manager.

### CODE REVIEW

**Comment by yasirfolio3** [comment]
@mikeng13 i have spotted an issue, The test's for polling manager defined here `https://github.com/optimizely/go-sdk/blob/4457b50f3d7da827143667b5dfafd07c1cc6d7bc/optimizely/config/polling_manager_test.go#L39` are not working correctly as NewDatafileProjectConfig method is returning nil config for the mock datafile. They are passing because `configManager.GetConfig` is also returning nil for this test.

**Comment by yasirfolio3** [comment]
> Can we add unit tests for config manager and check status of project config through these various scenarios?

Done, i have also fixed one of the previous tests which was producing nil project config due to unmarshalling issues, can you please review it now?

File: optimizely/config/polling_manager_test.go [hunk 1]
```diff
@@ -74,3 +76,57 @@ func TestNewPollingProjectConfigManagerWithNull(t *testing.T) {
 	_, err := configManager.GetConfig()
 	assert.NotNil(t, err)
 }
+
+func TestNewPollingProjectConfigManagerWithSimilarDatafileRevisions(t *testing.T) {
+	mockDatafile1 := []byte(`{"revision":"42","projectId":"1000"}`)
+	mockDatafile2 := []byte(`{"revision":"42","projectId":"1001"}`)
```
Comment (aliabbasrizvi) [inline optimizely/config/polling_manager_test.go]: Ah, this is interesting. I'd recommend using the same `projectId` or better create 2 project configs from the same mock datafile. So you can just have 1 variable `mockDatafile` and create `projectConfig1` and `projectConfig2` from it.

File: optimizely/config/polling_manager_test.go [hunk 2]
```diff
@@ -74,3 +76,57 @@ func TestNewPollingProjectConfigManagerWithNull(t *testing.T) {
...
+	assert.NotNil(t, actual)
+	assert.Equal(t, projectConfig1, actual)
+
+	configManager.SyncConfig(mockDatafile2)
+	actual, err = configManager.GetConfig()
+	assert.Equal(t, projectConfig1, actual)
+}
+
+func TestNewPollingProjectConfigManagerWithDifferentDatafileRevisions(t *testing.T) {
+	mockDatafile1 := []byte(`{"revision":"42","projectId":"1000"}`)
```
Comment (aliabbasrizvi) [inline optimizely/config/polling_manager_test.go]: Same as above.

File: optimizely/config/polling_manager_test.go (hunk 1 above)
Comment (yasirfolio3) [reply optimizely/config/polling_manager_test.go]: A slight change in both mock datafile's was required to verify that after calling `SyncConfig` using the second mock datafile, we still get project config generated for the first mock datafile. As per the suggestion, if we created 2 configs using the same mockDatafile, we won't be able to distinguish whether config for first datafile was returned or the second?

File: optimizely/config/polling_manager_test.go (hunk 2 above)
Comment (yasirfolio3) [reply optimizely/config/polling_manager_test.go]: I have updated the projectId for this test.

File: optimizely/config/polling_manager_test.go (hunk 1 above)
Comment (aliabbasrizvi) [reply optimizely/config/polling_manager_test.go]: My problem is with the premise of the test itself. It is an improbable event for project ID to be different. It does not line up with the mental model.

Why not pick a different field. Say set `botFiltering` to `False` in one datafile and to `True` in other datafile.

Comment (yasirfolio3) [reply optimizely/config/polling_manager_test.go]: Thanks, makes complete sense, i have made the suggested changes in the test-case.
